### PR TITLE
Coin filter bug fixes and improvements

### LINKED ISSUE #2083: Accounts: Filters
**[Zeplin](https://app.zeplin.io/project/5ee87c00f6af719a645d14c3/screen/5efc87a5cb1263131d55bd0d)** 

- [x] Add cross to remove search input 

Screen Shot 2020-08-15 at 22 54 26

- [x] Use default placeholder "_e.g. Search_" (as opposed to "Filter accounts") 

- [x] Use "_BTC_" (i.e.  the official 3 letter coin abbreviation) as active filter result (as opposed to "symbol:BTC")

Screen Shot 2020-08-15 at 22 56 38

### LINKED ISSUE #2319: Accounts, Add account/Find my accounts - No filter switch when other coin's new account is added
**Info:**
> Wallet Version: web, 20.9.1 cf1f180
OS: NixOS
Browser: any supported (tested Firefox 79.0)
Device: Model 1, 1.9.3, 0d5f00668

**Describe the bug**
The AccountSearchBox__CoinsFilter does not switch to the coin of the newly created account when a different coin than of the currently filtered is added (or undiscovered coin via Find my accounts button); discovered when testing the #2220 issue.

**Preconditions:**
- AccountSearchBox__CoinsFilter active on an arbitrary coin,
- 10 account limit is not exceeded on the other coin.

**Steps to reproduce:**
1. **activate filter** for an arbitrary coin (LTC for example) and select Add account
2. select a coin **other than the currently filtered** (BTC legacy for example) on the displayed modal
3. select **Add account** (or Find my accounts if adding a previously disabled coin) 

**Expected result:**
The AccountSearchBox__CoinsFilter switches to the coin of the newly added (or discovered) account's coin.

**Actual result:**
Filter deactivates for any previously selected coin.

**Screenshots:**
Add account Modal open, Litecoin filter selected:
image
Litecoin filter active after adding the BTC legacy account above:
image

### LINKED ISSUE #2556: Filter & Search: Empty states improvement
- [ ] Remove the coin icon when only 1 coin enabled: **[Zeplin](https://app.zeplin.io/project/5ee87c00f6af719a645d14c3/screen/5f7f305f0dd94258e3c916f3/)**

- [ ] Show empty state for no search results: **[Zeplin](https://app.zeplin.io/project/5ee87c00f6af719a645d14c3/screen/5f7f3060de903040bba7b176/)**

### PR DESCRIPTION
**Pull request by slowbackspace**
At first, the idea to use react context instead of full-blown redux looked really promising till I needed to fix some bugs where it was necessary to react to redux actions from bunch of places 🤕 I took it as an opportunity to rewrite it with state stored in redux.

New stuff (close https://github.com/trezor/trezor-suite/issues/2556):
- "No results" msg if there are, well... no results
- hide coin filter if only one coin is enabled

Coin filter and search string now resets on (fix https://github.com/trezor/trezor-suite/issues/2083 and fix https://github.com/trezor/trezor-suite/issues/2319): 
- disabling/enabling coins
- switching to another device/wallet
- adding a new account (handled directly in add-account modal as reacting to account.create would cause annoying resets during initial discovery)
- search string is reseted when coming from a click on a coin asset in the dashboard

First two are handled in accountSearch reducer. Another option would be to put it inside a middleware, but I didn't really want to create a whole new middleware just for this and adding it to existing wallet middleware seems to me confusing as well)

### CODE REVIEW

File: packages/suite/src/utils/wallet/accountUtils.ts [hunk 1]
```diff
@@ -462,15 +462,15 @@ export const getAccountIdentifier = (account: Account) => {
...
     coinFilter?: Account['symbol'],
 ) => {
     // if coin filter is active and account symbol doesn't match return false and don't continue the search
     const coinFilterMatch = coinFilter ? account.symbol === coinFilter : true;
     if (!coinFilterMatch) return false;
 
-    const searchString = rawSearchString.trim().toLowerCase();
-    if (searchString.length === 0) return true; // no search string
+    const searchString = rawSearchString?.trim().toLowerCase();
+    if (!searchString || searchString.length === 0) return true; // no search string
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/accountUtils.ts]: `!searchString` is enough, js check returns empty string as `false` so it covers both cases

File: packages/suite/src/utils/wallet/accountUtils.ts [hunk 2]
```diff
@@ -462,15 +462,15 @@ export const getAccountIdentifier = (account: Account) => {
 
 export const accountSearchFn = (
     account: Account,
-    rawSearchString: string,
+    rawSearchString: string | undefined,
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/accountUtils.ts]: `rawSearchString?: string,`

File: packages/suite/src/actions/wallet/accountSearchActions.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Account } from '@wallet-types';
+import { ACCOUNT_SEARCH } from '@wallet-actions/constants';
+
+export type AccountSearchActions =
+    | {
+          type: typeof ACCOUNT_SEARCH.SET_COIN_FILTER;
+          payload: Account['symbol'] | undefined;
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/accountSearchActions.ts]: `payload?: Account['symbol'];`

File: packages/suite/src/actions/wallet/accountSearchActions.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      }
+    | {
+          type: typeof ACCOUNT_SEARCH.SET_SEARCH_STRING;
+          payload: string | undefined;
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/accountSearchActions.ts]: `payload?: string;`

File: packages/suite/src/actions/wallet/accountSearchActions.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      };
+
+export const setCoinFilter = (payload: Account['symbol'] | undefined) => ({
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/accountSearchActions.ts]: `payload?: Account['symbol']`

File: packages/suite/src/actions/wallet/accountSearchActions.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    type: ACCOUNT_SEARCH.SET_COIN_FILTER,
+    payload,
+});
+
+export const setSearchString = (payload: string | undefined) => ({
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/accountSearchActions.ts]: `payload?: string`

File: packages/suite/src/components/wallet/AccountsMenu/components/AccountSearchBox/index.tsx [hunk 7]
```diff
@@ -31,6 +30,17 @@ const OuterCircle = styled.div<{ isSelected?: boolean; isMobile?: boolean }>`
...
 `;
 
+const InputWrapper = styled.div<{ showCoinFilter: boolean }>`
+    ${props =>
+        !props.showCoinFilter &&
+        css`
+            /* additional space under input if we are not showing coin filter */
+            /* one could think why not to remove a margin from coin filter so it can be here regardless of whether coin filter is shown */
+            /* but hold your horses, it is actually essential there is top PADDING on coin filter as a click to the area triggers deactivating the filter */
+            margin-bottom: 12px;
```
Comment (szymonlesisz) [inline packages/suite/src/components/wallet/AccountsMenu/components/AccountSearchBox/index.tsx:40]: :) :) :) nice comment :)

**Review by szymonlesisz — CHANGES_REQUESTED** [review]
code looks nice i didn't check if it's working tho, leaving this to qa :)

File: packages/suite/src/actions/wallet/accountSearchActions.ts (hunk 5 above)
Comment (slowbackspace) [reply packages/suite/src/actions/wallet/accountSearchActions.ts]: :D I don't know why I always subconsciously prefer this syntax. I will update it.

File: packages/suite/src/utils/wallet/accountUtils.ts (hunk 1 above)
Comment (slowbackspace) [reply packages/suite/src/utils/wallet/accountUtils.ts]: huh, TIL!

File: packages/suite/src/actions/wallet/accountSearchActions.ts (hunk 5 above)
Comment (slowbackspace) [reply packages/suite/src/actions/wallet/accountSearchActions.ts]: oh, there is a slight difference. Syntax I used won't allow calling setCoinFilter without an arg (`setCoinFilter ()` throws An argument for 'payload' was not provided), requested change will allow it just fine. I like that the argument needs to be passed even if it is undefined (it nicely tells a story that we are really setting it to `undefined`), but I don't have a strong opinion on it. Should I proceed updating the code to use `?`?

Comment (szymonlesisz) [reply packages/suite/src/actions/wallet/accountSearchActions.ts]: ah, ok so leave it as is then. it was only a nitpick

Comment (slowbackspace) [reply packages/suite/src/actions/wallet/accountSearchActions.ts]: too late! updated! 😁

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**